src/server/worker/api_server.go

```diff
@@ -1228,10 +1228,13 @@ func (a *APIServer) worker() {
 				}
 				return fmt.Errorf("error from InspectJob(%v): %+v", jobID, err)
 			}
-			if jobInfo.PipelineVersion != a.pipelineInfo.Version {
-				return fmt.Errorf("job's version (%d) doesn't match pipeline's "+
+			if jobInfo.PipelineVersion < a.pipelineInfo.Version {
+				continue
+			}
+			if jobInfo.PipelineVersion > a.pipelineInfo.Version {
+				return fmt.Errorf("job %s's version (%d) greater than pipeline's "+
```

Comment:
This is the other major bug fix. The issue here was that if you wind up with a job whose `PipelineVersion` is lower than the worker's current version then it'll go into backoff, expecting that the worker will be recreated soon and have the correct version. This works when the `PipelineVersion` is higher, but when it's lower we want to just skip the job.